### DISCUSSION TITLE
New builds Q1 2024

### DISCUSSION THREAD

**Discussion by ianw1974** [ianw1974/zimbra-build-scripts#67] [Announcements]
Coming in exactly 3 days will be the first builds of 2024!  These should address some of the CVE's Zimbra have recently fixed, mostly relating to the updating of Java.  A lot of these surprisingly have been in the CVE database since at least April 2022 if not longer in some cases for some of the issues fixed.

Updates will be posted here once the builds are completed and the site is updated with the new downloads.  Until then, hope you all had a great Christmas and enjoy your New Year celebrations!

**Comment by ianw1974** [comment]
Happy New Year to everyone!

Builds are now published and online ready to be downloaded.

**Comment by awnz** [comment]
Thanks Ian for all this work! Much appreciated.

I hit an issue using the Jan1 build to upgrade my Zimbra 8.8 installation to 9.0 - /opt/zimbra/conf/nginx/templates/nginx.conf.main.template seemed to update to what should be in Zimbra 10, not Zimbra 9.

I described what effect that had on the Zimbra forum [here](https://forums.zimbra.org/viewtopic.php?p=312421#p312421) - in brief, nginx ran as nobody instead of zimbra, couldn't access /opt/zimbra/data/tmp/nginx/proxy/ which in turn caused attachment downloads and a few other things to randomly fail.

This is using the Zimbra 9.0.0 Ubuntu 18.04 build. I've just redownloaded it and examined zimbra-proxy_9.0.0.GA.0124.UBUNTU18.64_amd64.deb > data.tar.xz > opt/zimbra/conf/nginx/templates and can confirm it's missing these three lines:

```
# change UID/GID to zimbra/zimbra upon startup
# 
user ${main.user} ${main.group};
```
I've fixed the issue by adding the missing lines for my Zimbra install.

This is as far as I've traced where the issue came from, do you know what should be checked next to see where this wrong file came from?

Cheers

**Comment by ianw1974** [comment]
Hi,

Yeah I saw that on the forum yesterday I think.  I can patch that during build as I also do with some Jetty stuff as well due to OnlyOffice not being in v9.  So that will solve that anyway.

Since 10.0.6, most people have been migrating their installs from 8.x to 10.x purely due to the fact Zimbra 9 goes EOL far quicker.  Prior to 10.0.6 there wasn't a clear upgrade path, but I believe that has been addressed by Synacor/Zimbra hence why 10.0.6 works now for upgrades from earlier versions.

Ideally your path forward is upgrade your 9.x to 10.x now, but in the meantime I will patch the affected file for v9 versions with the lines that are missing.  At least that way it'll solve the problem.

FYI - I build from ```develop``` purely because there is no real way to obtain an automated list of tags.  Not all repos are tagged with 10.0.6 or whatever, it just gets tagged when changes are made across the 80+ repositories that Zimbra has.  If there was a published list, or at least a single Zimbra repository that did contain a concise list of tags, then I would offer the ability to build this way.  That said, people have attempted builds using tags to say build 10.0.6 and had far more problems with their build than for example my 10.0.6 build from develop branch.  My viewpoint is, building from develop we get the absolute latest, which means it is at least 10.0.6 or the latest 9.0.0 available than something that is a snapshot in the past and potentially problematic.  Some will agree/disagree with this.  I do have an open issue for the tags, but I don't plan to address this until a concise list of tags can be obtained in an automated way.

Until then, the easiest is to do as you have done and report the issue, so I can patch accordingly and the problem is resolved.  And you also have the absolute latest 9.x release.  There potentially could be other issues in the future that also need patching, or perhaps even some we haven't come across yet, but as long as they get fixed quickly and effectively as I have been doing so far when things have been reported, I think there's nothing much wrong with that.  Bugs will happen on software :)

I have opened an issue for this, you can follow here: #72

**Comment by awnz** [comment]
Thanks Ian :)

**Comment by jeredfloyd** [comment]
Reporting success updating from 10.0.5 to 10.0.6; no problems encountered.
